storage/rocksdb/ha_rocksdb.cc

```diff
@@ -11832,12 +11818,19 @@ bool ha_rocksdb::skip_unique_check() const {
          any indexes. If the table has secondary keys, then those might becomes
          inconsisted/corrupted
       3) We're using read-free replication
+      4) For CREATE INDEX, DROP INDEX, & ALTER TABLE when any duplicate key
+         violations are user's responsibility
   */
-  return THDVAR(table->in_use, bulk_load) ||
-         (my_core::thd_test_options(table->in_use,
-                                    OPTION_RELAXED_UNIQUE_CHECKS) &&
+  auto *const thd = table->in_use;
+  return THDVAR(thd, bulk_load) ||
+         (my_core::thd_test_options(thd, OPTION_RELAXED_UNIQUE_CHECKS) &&
           m_tbl_def->m_key_count == 1) ||
-         use_read_free_rpl();
+      use_read_free_rpl() ||
+      ((table_share->table_category == TABLE_CATEGORY_TEMPORARY ||
```

Comment:
looks like it is easy to distinguish user temp table vs intrinsic temp table..  if it is easy, maybe user temp table could skip unique check but intrinsic temp table still use unique check

Reply:
By code comments the user temp tables are `TABLE_CATEGORY_USER` and internal temp tables are `TABLE_CATEGORY_TEMPORAY`. So it's easy to tell temp tables apart, but harder to tell user regular and temporary tables apart.

To address your comment, I believe it is enough to remove `TABLE_CATEGORY_TEMPORARY`.

Reply:
sadly, that code comments maybe is misunderstanding or incorrect. Just create a temporarily table and debugger shows that user temp tables are TABLE_CATEGORY_TEMPORAY

Callstack:
init_tmp_table_share // share->table_category = TABLE_CATEGORY_TEMPORARY;
ha_create_table
rea_create_tmp_table
create_table_impl
mysql_create_table_no_lock
mysql_create_table
Sql_cmd_create_table::execute

Reply:
anyway, remove TABLE_CATEGORY_TEMPORARY looks good to me.